src/main/scala/com/example/solution/domain/room.scala

```diff
@@ -0,0 +1,14 @@
+package com.example.solution.domain
+
+object room {
+
+  case class RoomId(id : Int) extends AnyVal
+  case class Room(
+                 id : RoomId,
+                 rows : Int,
```

Comment:
It can't, it's pretty sloppy, we could have for instance use Refined library to check if a value is positive at compile time, or even create a smart constructor that would check if the provided integer is positive.